src/connection_string.ts

```diff
@@ -347,10 +347,10 @@ export function parseOptions(
     allProvidedOptions.set(key, values);
   }
 
-  if (
+  const didMapTLSCertificateFile =
     allProvidedOptions.has('tlsCertificateKeyFile') &&
-    !allProvidedOptions.has('tlsCertificateFile')
-  ) {
+    !allProvidedOptions.has('tlsCertificateFile');
+  if (didMapTLSCertificateFile) {
     allProvidedOptions.set('tlsCertificateFile', allProvidedOptions.get('tlsCertificateKeyFile'));
   }
```

Comment:
Possibly missing something here, why are we still setting the deprecated option? can't we change this to set `tlsCertificateKeyFile`

Reply:
If we hit this logic, `tlsCertificateKeyFile` is already set.  This logic was explicitly added in https://jira.mongodb.org/browse/NODE-3591

The bug here is specifically when the user sets `tlsCertificateKeyFile` and not `tlsCertificateFile `, because we assign `tlsCertificateKeyFile` to `tlsCertificateFile ` and then check if the user has set a deprecated option.

I'm not sure what would happen if just removed this logic but I'm guessing we'd reintroduce the bug Daria fixed in 3591.  Ultimately, I'd rather just fix it like this because this is the 5.x branch and the logic is removed entirely from 6.x.